compiler-plugin/src/main/kotlin/arrow/meta/quotes/BinaryOper.kt

```diff
@@ -0,0 +1,26 @@
+package arrow.meta.quotes
+
+import arrow.meta.Meta
+import arrow.meta.phases.ExtensionPhase
+import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.psi.KtBinaryExpression
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtOperationReferenceExpression
+
+/**
+ * A template destructuring [Scope] for a [KtBinaryExpression]
+ */
+fun Meta.binaryOper(
```

Comment:
I think the name for this quote and all other quotes should be whatever the Psi element name is.
I think we should replace:
```
func -> namedFunction
bynaryOper -> binaryExpression
```
there are other function matches we can do like lambdas and there are other binary expressions that are not operators for example any function flagged as `infix`

Reply:
great! I'll have this fixed, and also add to the AST parsing issue as a note!